engine/v2/transaction/record/paymail.go

```diff
@@ -0,0 +1,128 @@
+package record
+
+import (
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/spv-wallet/conv"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/errors"
+	"github.com/bitcoin-sv/spv-wallet/models/transaction/bucket"
+)
+
+type paymailInfo struct {
+	annotation       *transaction.PaymailAnnotation
+	vouts            map[uint32]struct{}
+	skipNotification bool
+}
+
+func newPaymailInfo() paymailInfo {
+	return paymailInfo{
+		annotation:       nil,
+		vouts:            map[uint32]struct{}{},
+		skipNotification: false,
+	}
+}
+
+func (pi *paymailInfo) empty() bool {
+	return pi.annotation == nil
+}
+
+func (pi *paymailInfo) equalsToAnnotation(annotation *transaction.PaymailAnnotation) bool {
+	return *pi.annotation == *annotation
+}
+
+func (pi *paymailInfo) add(vout int, annotation *transaction.PaymailAnnotation) error {
+	if !pi.empty() && !pi.equalsToAnnotation(annotation) {
+		return txerrors.ErrMultiPaymailRecipientsNotSupported
+	}
+
+	vout32, err := conv.IntToUint32(vout)
+	if err != nil {
+		return txerrors.ErrAnnotationIndexConversion.Wrap(err)
+	}
+
+	pi.annotation = annotation
+	pi.vouts[vout32] = struct{}{}
+	return nil
+}
+
+func (pi *paymailInfo) hasVOut(vout uint32) bool {
+	if pi.empty() {
+		return false
+	}
+	_, ok := pi.vouts[vout]
+	return ok
+}
+
+func (pi *paymailInfo) Sender() string {
+	if pi.empty() {
+		return ""
+	}
+	return pi.annotation.Sender
+}
+
+func (pi *paymailInfo) Receiver() string {
+	if pi.empty() {
+		return ""
+	}
+	return pi.annotation.Receiver
+}
+
+func (pi *paymailInfo) Reference() string {
+	if pi.empty() {
+		return ""
+	}
+	return pi.annotation.Reference
+}
+
+func (f *txFlow) processPaymailOutputs(annotations transaction.Annotations) (paymailInfo, error) {
+	info := newPaymailInfo()
+
+	for vout, annotation := range annotations.Outputs {
```

Comment:
Do we have a nil check for annotations somewhere? 
I don't see it under `RecordTransactionOutline`.

Reply:
You are right, closing